code/game/objects/structures/divine.dm

```diff
@@ -1,3 +1,20 @@
+GLOBAL_LIST_INIT(potion_rates, list(
+	/obj/item/slimepotion/slime/sentience		=9,
+	/obj/item/slimepotion/speed		=2,
+	/obj/item/slimepotion/slime/renaming			=9,
+	/obj/item/slimepotion/slime/slimeradio		=9,
+	/obj/item/slimepotion/peacepotion		=9
+))
+
+GLOBAL_LIST_INIT(gunparts_rates, list(
```

Comment:
I'd prefer if instead of a fixed list for all guns, something to make it based off the quality or rarity of the gun. Like, basic hoboguns would give at most a couple parts, while an assault rifle would give some serious parts, and a minigun or gauss rifle would basically explode into valuables.

Also, might want to check out the crafting recipes for various guns, and make sure it won't end up with a net positive. At least not a reliable one, if its chance based whether it'll actually happen, that should be fine.

Reply:
The basic idea of ​​sacrificing weapons is that they melt right on the altar, and you have to be very lucky to extract at least something useful from the remains. In any case, the loot obtained in this way does not exceed the cost of crafting any weapon, so this system will not be abused.

Reply:
Fair enough, we'll see how it works out and make changes later if needed.